connection.go

```diff
@@ -247,12 +250,18 @@ func (ch *Channel) newConnection(conn net.Conn, initialState connectionState, ev
 	}
 	c.inbound.onRemoved = c.checkExchanges
 	c.outbound.onRemoved = c.checkExchanges
+	c.inbound.onAdded = c.onExchangeAdded
```

Comment:
can this direcltly call `c.callOnExchangeChange`? and then we can remove `onExchangeAdded`